scripts/qa_crowbarsetup.sh

```diff
@@ -1956,23 +1941,69 @@ function proposal_set_value
     proposal_modify_value "$1" "$2" "$3" "$4" "="
 }
 
+# get the currently set value in the proposal
+function proposal_get_value
+{
+    local proposal="$1"
+    local proposaltype="$2"
+    local variable="$3"
+
+    local pfile=`get_proposal_filename "${proposal}" "${proposaltype}"`
+
+    safely rubyjsonparse "puts j${variable}" < $pfile
+}
+
 # wrapper for proposal_modify_value
 function proposal_increment_int
 {
     proposal_modify_value "$1" "$2" "$3" "$4" "+="
 }
 
+function setup_trusted_cert
+{
+    local certfile=$1
+    local keyfile=$2
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+        local clusternodes_var=$(echo clusternodes${clusternameservices})
+        local controller_nodes=${!clusternodes_var}
+        local cn=$(get_cluster_vip_hostname $clusternameservices)
+    else
+        local controller_nodes=($(get_all_discovered_nodes | head -n 1))
+        local cn=$controller_nodes
+    fi
+
+    if [ ! -f $cn.key ] ; then
+        subj_alt="DNS:${cn},DNS:public-${cn}"
```

Comment:
Just curious, what's "DNS:public-${cn}" for? I have not came across this convention before.

Reply:
Crowbar by default names its public interface "public-$fqdn". I don't think we ever use that in CI but I added it for completeness in case we ever do.